android/src/main/java/com/reactnativestripesdk/StripeSdkModule.kt

```diff
@@ -595,41 +607,52 @@ class StripeSdkModule(private val reactContext: ReactApplicationContext) : React
     val factory = PaymentMethodCreateParamsFactory(setupIntentClientSecret, params, cardFieldView, cardFormView)
 
     try {
-      val activity = currentActivity as ComponentActivity
       val confirmParams = factory.createSetupParams(paymentMethodType)
-      stripe.confirmSetupIntent(activity, confirmParams)
+      (currentActivity as? ComponentActivity)?.let {
+        stripe.confirmSetupIntent(it, confirmParams)
+      } ?: run {
+        promise.resolve(createMissingActivityError())
```

Comment:
When is it possible for currentActivity to be `null`?

Reply:
Please forgive my lack of experience with Kotlin code, is `currentActivity` a dynamic getter here where it might become non-null on future invocations? or is `currentActivity` a member variable that is captured by the React-Native-Module? (If it's the latter, then retrying wouldn't fix the problem)

Additionally, it feels like _very early_ in startup is not an expected time to call the Stripe SDK

Reply:
it's dynamic, react modules should NOT have any long lived reference to activities